### ISSUE TITLE
Use existing extension selection box for the history search field, preventing user error

### ISSUE THREAD

**Issue by hexylena** [galaxyproject/galaxy#14294]
Testing 22.05, edited given discussion + splitting out of https://github.com/galaxyproject/galaxy/issues/14305)

The new history search form prompted me to filter by "extension", so, naturally I filtered by file extension (.tsv), which I didn't immediately pick up to mean the internal datatype extension.

If there could be a dropdown, like we use in the "Change Datatype" menu, I think users would benefit from that.

![image](https://user-images.githubusercontent.com/458683/178301133-275eba51-b6d6-4224-8fcb-bb03cae6702c.png)
 

![image](https://user-images.githubusercontent.com/458683/178301036-b1ff2a10-0171-49cb-b087-5feac4f32dca.png)

**Comment by hexylena** [comment]
Yes, exactly, it's `tabular`, not tsv the specific `tsv` datatype, hence search returning zero results for `extension:tsv` which I did when reading "filter on file extension".

The verbiage used is the potentially confusing part to end users, who don't know that "extension" here doesn't mean "extension" as they've come to know, but rather "datatype". Just renaming that to `datatype` would remove that confusion I think.

**Comment by mvdbeek** [comment]
But it's not the datatype technically, it **is** the extension. Users should absolutely not put the extension in the file name and that would be something to work on, e.g stripping these by default. I think this would actually be more confusing.

**Comment by hexylena** [comment]
Sure, then we need to unify the other way, at minimum stripping user provided extensions and maybe (uneditably) showing the extension in the dataset name in history?

But internally we call this "datatype", or have I got something completely wrong? 

Either way on a single term would be ideal:

- In the upload dialog it's called "type" (here I read 'datatype')
- Dataset edit/convert menu calls it "datatype"

"datatype" as a label makes a lot more sense to me personally than "extension" which I really only associate with the filename on disk, not the file's internal format, but I'm happy either way.

**Comment by mvdbeek** [comment]
> `extension` isn't really exposed anywhere then right?

It sure is, in the UI, the API, the model. I'm not sure how you reached the conclusion

> if it's the `file_ext` attribute of the datatype classes that you mean,

it is not, that's just an alias that should go away IMO.

You just can't gloss this over, I think this proposition is just not feasible.

**Comment by hexylena** [comment]
> It sure is, in the UI, the API, the model. I'm not sure how you reached the conclusion

Could you give an example of this then in the UI? I seem to be missing something very key then.

**Comment by hexylena** [comment]
I'm struggling a bit with:  If I don't understand the distinction you're making between datatype vs extension, as an admin and (low-experience) galaxy dev, how is an end user supposed to understand what goes in this field, especially when there's a common usage of 'extension' that they might know and would be misled by. (And I believe I'm being misled by, given your responses.)

**Comment by mvdbeek** [comment]
I mean, format, extension, file_ext ... these are all fine, even if they should be just one. extension is probably the right thing to choose, but I don't care which one to pick. What is not OK is to mix this with the dataset name, as per your request. It also literally maps to the file extension you get on download or when we reconstitute a disk name, so ... I don't really understand how this discussion related to the issue, which is people putting the extension in the file name. The moment you run a couple of tools in Galaxy you'll noticed this doesn't make sense, cause Galaxy already has the extension.

Each datatype has a extension, selecting one per extension makes sense, doesn't it ? But datatype is more than just extension, so it doesn't make sense to set an extension, you're setting a datatype. This is what I don't think we can fix without making things more confusing.

**Comment by mvdbeek** [comment]
And here's a selection of extensions, are what should be extensions:

![Screenshot 2022-07-11 at 17 48 11](https://user-images.githubusercontent.com/6804901/178294435-bb414fe2-c03f-455a-8ac5-134d0bb7bbc8.png)
![Screenshot 2022-07-11 at 17 48 55](https://user-images.githubusercontent.com/6804901/178294439-7798fddc-950a-4c58-9f79-8279da71d73c.png)
![Screenshot 2022-07-11 at 17 49 23](https://user-images.githubusercontent.com/6804901/178294443-faeb6707-533e-4844-8078-b8b1ee48dc83.png)
![Screenshot 2022-07-11 at 17 49 54](https://user-images.githubusercontent.com/6804901/178294446-761513f6-2cfa-4d24-8b43-1417987cc6de.png)
![Screenshot 2022-07-11 at 17 50 32](https://user-images.githubusercontent.com/6804901/178294447-bfc97d58-665a-40b6-8bc6-d582efc823e3.png)

**Comment by hexylena** [comment]
> And here's a selection of extensions, are what should be extensions:

Thanks, those pictures really help! I guess sometimes the line between extension and datatype blurs for me, especially if it's a 1:1 mapping of every datatype having an extension, and some places we refer to the type or datatype (like in upload) and others the format (dataset peek, seems to be extension), and the library's "details" (also extension).

_Thank you for your patience_, explaining in a bit more detail. I was really conceptually missing what you meant by `extension`. 

> What is not OK is to mix this with the dataset name, as per your request

Not what I was requesting, honest.

> so ... I don't really understand how this discussion related to the issue, which is people putting the extension in the file name

_We're on the same page_, the extension doesn't belong in the filename. (When I made the mid-thread comment, it was more a soft proposal to do something like the following image, to help keep the datatypes closer to filename like users might be expecting after their uploads, but again, **I don't really care about this**, my issue was about UI terminology confusion.)
![image](https://user-images.githubusercontent.com/458683/178295482-2ae3f61d-5e8e-491b-b8e1-1d5d052ed9e1.png)

> Each datatype has a extension, selecting one per extension makes sense, doesn't it ? But datatype is more than just extension, so it doesn't make sense to set an extension, you're setting a datatype.

**Checking my understanding:** So in the interface we're always showing a list of extensions, _not datatypes_, but just labelling them type or datatypes, or format or 'extension' in the libraries.

---

**Clarifying the original issue**:

1. I was confused by 'extension:' in search, that the field isn't called 'datatype' since 'datatype' is used so commonly to refer to the extension within the UI (but this is more about using [Type, Datatype, Format, Extension, Details] to all point to the same thing: the extension.)
2. And proposing that this interface could be improved significantly through a selection box like we use in change datatype's interface which lists the possible options.

**Comment by mvdbeek** [comment]
> **Checking my understanding:** So in the interface we're always showing a list of extensions, _not datatypes_, but just labelling them type or datatypes, or format or 'extension' in the libraries.

That how I understand it too, yes.

> 2\. And proposing that this interface could be improved significantly through a selection box like we use in change datatype's interface which lists the possible options.

That is definitely a papercut we can do, got everything available in the store.

**Comment by hexylena** [comment]
Would it be useful to file an issue tracking the different locations/verbiage used across the interface to refer to extension? And then we could pick one and standardise? Or is it relevant/useful/important to call it different things in different locations?
